code/modules/robotics/bot/secbot.dm

```diff
@@ -803,16 +809,11 @@
 	// look for a criminal in range of the bot
 	proc/look_for_perp()
 		src.anchored = 0
-		for_by_tcl(C, /mob/living/carbon) //Let's find us a criminal
-			if(!IN_RANGE(src, C, 7)) // We've made a plea bargain with opaque objects to turn in criminals hiding behind them
-				continue
+		for(var/mob/living/carbon/C in view(7, get_turf(src))) //Let's find us a criminal
```

Comment:
GET_NEARBY ?

Reply:
That'd make them able to see through walls, which I'm trying to fix.